gateway-ha/src/main/java/io/trino/gateway/ha/handler/ProxyUtils.java

```diff
@@ -0,0 +1,164 @@
+ */
+package io.trino.gateway.ha.handler;
+
+import com.google.common.base.Splitter;
+import com.google.common.io.CharStreams;
+import io.airlift.log.Logger;
+import jakarta.servlet.http.HttpServletRequest;
+
+import java.io.InputStreamReader;
+import java.util.Base64;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.trino.gateway.ha.handler.QueryIdCachingProxyHandler.TRINO_UI_PATH;
+import static io.trino.gateway.ha.handler.QueryIdCachingProxyHandler.USER_HEADER;
+import static io.trino.gateway.ha.handler.QueryIdCachingProxyHandler.V1_QUERY_PATH;
+import static io.trino.gateway.ha.handler.QueryIdCachingProxyHandler.V1_STATEMENT_PATH;
+
+public final class ProxyUtils
+{
+    public static final String SOURCE_HEADER = "X-Trino-Source";
+    public static final String AUTHORIZATION = "Authorization";
+
+    private static final Logger log = Logger.get(ProxyUtils.class);
+    public static final int QUERY_TEXT_LENGTH_FOR_HISTORY = 200;
+    /**
+     * This regular expression matches query ids as they appear in the path of a URL. The query id must be preceded
+     * by a "/". A query id is defined as three groups of digits separated by underscores, with a final group
+     * consisting of any alphanumeric characters.
+     */
+    private static final Pattern QUERY_ID_PATH_PATTERN = Pattern.compile(".*/(\\d+_\\d+_\\d+_\\w+).*");
+    /**
+     * This regular expression matches query ids as they appear in the query parameters of a URL. The query id is
+     * defined as in QUERY_TEXT_LENGTH_FOR_HISTORY. The query id must either be at the beginning of the query parameter
+     * string, or be preceded by %2F (a URL-encoded "/"), or  "query_id=", with or without the underscore and any
+     * capitalization.
+     */
+    private static final Pattern QUERY_ID_PARAM_PATTERN = Pattern.compile(".*(?:%2F|(?i)query_?id(?-i)=|^)(\\d+_\\d+_\\d+_\\w+).*");
+    private static final Pattern EXTRACT_BETWEEN_SINGLE_QUOTES = Pattern.compile("'([^\\s']+)'");
+
+    private ProxyUtils() {}
+
+    public static String getQueryUser(String userHeader, String authorization)
+    {
+        if (!isNullOrEmpty(userHeader)) {
+            log.debug("user from %s", USER_HEADER);
+            return userHeader;
+        }
+
+        log.debug("user from basic auth");
+        String user = "";
+        if (authorization == null) {
+            log.debug("didn't find any basic auth header");
+            return user;
+        }
+
+        int space = authorization.indexOf(' ');
+        if ((space < 0) || !authorization.substring(0, space).equalsIgnoreCase("basic")) {
+            log.error("basic auth format is incorrect");
+            return user;
+        }
+
+        String headerInfo = authorization.substring(space + 1).trim();
+        if (isNullOrEmpty(headerInfo)) {
+            log.error("The encoded value of basic auth doesn't exist");
```

Comment:
```suggestion
            log.error("Encoded value of basic auth doesn't exist");
```

but also .. not sure .. can we improve?